clients/python-wrapper/lakefs/branch.py

```diff
@@ -354,13 +354,19 @@ def __enter__(self):
         self._tx_branch = Branch(self._repo_id, self._tx.id, self._client)
         return self._tx
 
-    def __exit__(self, exc_type, exc_val, exc_tb):
-        try:
-            if len(list(self._tx.uncommitted(amount=1))) == 0:
-                return  # Do nothing if tx is empty
+    def __exit__(self, typ, value, traceback) -> bool:
+        if typ is not None:  # Do nothing in case exception occurred
+            return False  # Raise the underlying exception
+
+        if len(list(self._tx.uncommitted(amount=1))) == 0:  # Delete branch and raise exception
```

Comment:
In my previous comment, I meant without the need to list uncommitted, just react to the type of error returned when trying to commit and merge. Theoretically, someone can delete the changes from the branch between the list and commit, and we will still return no changes and not remove the branch